test/TensorFlowRuntime/tensor_group.swift

```diff
@@ -0,0 +1,230 @@
+  let simple: Simple
+  // Mutable.
+  var mixed: Mixed
+}
+
+struct Generic<T: TensorGroup & Equatable, U: TensorGroup & Equatable> : TensorGroup, Equatable {
+  var t: T
+  var u: U
+}
+
+TensorGroupTests.test("EmptyTypeList") {
+  expectEqual([], Empty._typeList)
+}
+
+TensorGroupTests.test("SimpleTypeList") {
+  let float = Float.tensorFlowDataType
+  expectEqual([float, float], Simple._typeList)
+}
+
+TensorGroupTests.test("SimpleInit") {
+  let w = Tensor<Float>(0.1)
+  let b = Tensor<Float>(0.1)
+  let simple = Simple(w: w, b: b)
+
+  let status = TF_NewStatus()
+  let wHandle = TFE_TensorHandleCopySharingTensor(w.handle._cTensorHandle, status)!
+  let bHandle = TFE_TensorHandleCopySharingTensor(b.handle._cTensorHandle, status)!
+  TF_DeleteStatus(status)
+
+  let buffer = UnsafeMutableBufferPointer<CTensorHandle>.allocate(capacity: 2)
+  buffer.baseAddress?.initialize(to: wHandle)
+  buffer.baseAddress?.advanced(by: 1).initialize(to: bHandle)
+  let expectedSimple = Simple(_owning: UnsafePointer(buffer.baseAddress))
+
+  expectEqual(expectedSimple, simple)
+}
+
+TensorGroupTests.test("MixedTypeList") {
+  let float = Float.tensorFlowDataType
+  let int = Int32.tensorFlowDataType
+  expectEqual([float, int], Mixed._typeList)
+}
+
+TensorGroupTests.test("MixedInit") {
+  let float = Tensor<Float>(0.1)
+  let int = Tensor<Int32>(1)
+  let mixed = Mixed(float: float, int: int)
+
+  let status = TF_NewStatus()
+  let floatHandle = TFE_TensorHandleCopySharingTensor(float.handle._cTensorHandle, status)!
+  let intHandle = TFE_TensorHandleCopySharingTensor(int.handle._cTensorHandle, status)!
+  TF_DeleteStatus(status)
+
+  let buffer = UnsafeMutableBufferPointer<CTensorHandle>.allocate(capacity: 2)
+  buffer.baseAddress?.initialize(to: floatHandle)
+  buffer.baseAddress?.advanced(by: 1).initialize(to: intHandle)
+  let expectedMixed = Mixed(_owning: UnsafePointer(buffer.baseAddress))
+
+  expectEqual(expectedMixed, mixed)
+}
+
+TensorGroupTests.test("NestedTypeList") {
+  let float = Float.tensorFlowDataType
+  let int = Int32.tensorFlowDataType
+  expectEqual([float, float, float, int], Nested._typeList)
+}
+
+TensorGroupTests.test("NestedInit") {
+  let w = Tensor<Float>(0.1)
+  let b = Tensor<Float>(0.1)
+  let simple = Simple(w: w, b: b)
+  let float = Tensor<Float>(0.1)
+  let int = Tensor<Int32>(1)
+  let mixed = Mixed(float: float, int: int)
+  let nested = Nested(simple: simple, mixed: mixed)
+
+  let status = TF_NewStatus()
+  let wHandle = TFE_TensorHandleCopySharingTensor(w.handle._cTensorHandle, status)!
+  let bHandle = TFE_TensorHandleCopySharingTensor(b.handle._cTensorHandle, status)!
+  let floatHandle = TFE_TensorHandleCopySharingTensor(float.handle._cTensorHandle, status)!
+  let intHandle = TFE_TensorHandleCopySharingTensor(int.handle._cTensorHandle, status)!
+  TF_DeleteStatus(status)
+
+  let buffer = UnsafeMutableBufferPointer<CTensorHandle>.allocate(capacity: 4)
+  buffer.baseAddress?.initialize(to: wHandle)
```

Comment:
`UnsafeMutableBufferPoninter` has its own [`initialize(from:)`](https://developer.apple.com/documentation/swift/unsafemutablebufferpointer), which lets you initialize all four elements from a sequence. 

```swift
buffer.initialize(from: [wHandle, bHanble, floatHandle, intHandle])
```

Same for other occurrences.